src/sc/sidechainTxsCommitmentBuilder.cpp

```diff
@@ -0,0 +1,177 @@
+
+    unsigned int ccOutIdx = 0;
+    LogPrint("sc", "%s():%d -getting leaves for vsc out\n", __func__, __LINE__);
+    for(; ccOutIdx < tx.GetVscCcOut().size(); ++ccOutIdx)
+    {
+        uint256 scId = tx.GetVscCcOut().at(ccOutIdx).GetScId();
+        sScIds.insert(scId);
+
+        std::vector<field_t*>& vec = mScMerkleTreeLeavesFt[scId]; //create or pick entry for scId
+        field_t* pField = mapScTxToField(tx.GetVscCcOut()[ccOutIdx].GetHash(), tx.GetHash(), ccOutIdx);
+        vec.push_back(pField);
+    }
+
+    LogPrint("sc", "%s():%d -getting leaves for vft out\n", __func__, __LINE__);
+    for(; ccOutIdx < tx.GetVftCcOut().size(); ++ccOutIdx)
+    {
+        uint256 scId = tx.GetVftCcOut().at(ccOutIdx).GetScId();
+        sScIds.insert(scId);
+
+        std::vector<field_t*>& vec = mScMerkleTreeLeavesFt[scId]; //create or pick entry for scId
+        field_t* pField = mapScTxToField(tx.GetVftCcOut()[ccOutIdx].GetHash(), tx.GetHash(), ccOutIdx);
+        vec.push_back(pField);
+    }
+
+    LogPrint("sc", "%s():%d - nIdx[%d]\n", __func__, __LINE__, ccOutIdx);
+}
+
+void SidechainTxsCommitmentBuilder::add(const CScCertificate& cert)
+{
+    uint256 scId = cert.GetScId();
+    sScIds.insert(scId);
+    mScCerts[scId] = mapCertToField(cert.GetHash());
+}
+
+uint256 SidechainTxsCommitmentBuilder::getCommitment()
+{
+    std::vector<field_t*> vSortedScLeaves;
+
+    // set of scid is ordered
+    for (const auto& scid : sScIds)
+    {
+        field_t* ftRootField = nullptr;
+        auto itFt = mScMerkleTreeLeavesFt.find(scid);
+        if (itFt != mScMerkleTreeLeavesFt.end() )
+        {
+            ftRootField = merkleTreeRootOf(itFt->second);
+        }
+
+        field_t* btrRootField = nullptr;
+        auto itBtr = mScMerkleTreeLeavesBtr.find(scid);
+        if (itBtr != mScMerkleTreeLeavesBtr.end() )
+        {
+            btrRootField = merkleTreeRootOf(itBtr->second);
+        }
+
+        field_t* certRootField = nullptr;
+        auto itCert = mScCerts.find(scid);
+        if (itCert != mScCerts.end() )
+        {
+            certRootField = itCert->second;
+        }
+
+        field_t* scIdField = mapCertToField(scid);
+
+        std::vector<field_t*> singleSidechainComponents = {ftRootField, btrRootField, certRootField, scIdField};
+        field_t * sidechainTreeRoot = merkleTreeRootOf(singleSidechainComponents);
+
+        vSortedScLeaves.push_back(sidechainTreeRoot);
+    }
+
+    field_t * finalTreeRoot = emptyField.zero;
+    if (vSortedScLeaves.size() != 0)
+        finalTreeRoot = merkleTreeRootOf(vSortedScLeaves);
+
+    uint256 res = mapFieldToHash(finalTreeRoot);
+    if (finalTreeRoot != emptyField.zero) zendoo_field_free(finalTreeRoot);
+
+    return res;
+}
+#endif
+
+field_t* SidechainTxsCommitmentBuilder::mapScTxToField(const uint256& ccoutHash, const uint256& txHash, unsigned int outPos) const
+{
+    static_assert((sizeof(uint256) + sizeof(uint256) + sizeof(unsigned int)) < SC_FIELD_SIZE);
```

Comment:
the syntax of static_assert used here compiles only from c++17 on, c+11 wants an error msg too:

static_assert ( bool_constexpr , message )    (since C++11)
static_assert ( bool_constexpr  )                    (since C++17)